### PR TITLE
Name the backbones according to the `explicit-type-name` extension (for R4)

### PR DESCRIPTION
**Pull request by ewoutkramer**
Since R3 quite a few backbones carry an explicit name, given in the "http://hl7.org/fhir/StructureDefinition/structuredefinition-explicit-type-name" extension. Use this name for these types, including in the generated TypeName property.

### CODE REVIEW

**Comment by brianpos** [comment]
@ewoutkramer   Make sure this one gets into the release notes, as there could be folks like me that could be effected by this one...
(I use it in some interim storage in some places - and I thought this was already used by the code generator)

**Comment by ewoutkramer** [comment]
It was already used by the code generator, but the name "Component" was added for the C# type name (fine), but also for the real "FHIR" typename (not what I wanted now).  And to make them unique, I have added the resource name before them.

To conclude: the names for the C# types do not change, but the name of the TypeName changes to say  "Patient#Contact"  (where it was ContactComponent before).

The whole reason for this is that these type names now surface due to the FhirPath normative `type()` function, you can call this on `Patient.contact` too, and then what will appear as the name?

I discussed this with Bryn, and we are working towards the situation where our API and the CQL spec align.

**Comment by brianpos** [comment]
Was that for the R5 stuff?
Each release I had to go in and tweak through that, I thought this was the name that James does in the HAPI libs too (using what we had, and likely gets this issue too)

**Comment by ewoutkramer** [comment]
No, this is for the FhirPath normative stuff. And we really need this to be unique, hence the prefix.

**Comment by ewoutkramer** [comment]
This will be part of the https://github.com/FirelyTeam/fhir-net-api/wiki/Breaking-changes-in-2.0 page!

**Comment by brianpos** [comment]
Then yes, definitely include in the release notes, Is this also effecting the classname generated?
(if not then note quite so bad as won't impact compile time - they should be unique anyway)

**Comment by ewoutkramer** [comment]
Should we include James in this discussion too?

**Comment by brianpos** [comment]
HAPI will have the same issues (or maybe Grahame knows about it too - don't recall if he wrote that part of the JAVA stack or not)

**Comment by ewoutkramer** [comment]
> Then yes, definitely include in the release notes, Is this also effecting the classname generated?
> (if not then note quite so bad as won't impact compile time - they should be unique anyway)

No, the .NET classnames remain the same.

File: src/Hl7.Fhir.Core/Model/Source/search-parameters.xml [hunk 1]
```diff
@@ -13159,7 +13159,8 @@
         <xpath value="f:Composition/f:encounter | f:DeviceRequest/f:encounter | f:DiagnosticReport/f:encounter | f:DocumentReference/f:context/f:encounter | f:Flag/f:encounter | f:List/f:encounter | f:NutritionOrder/f:encounter | f:Observation/f:encounter | f:Procedure/f:encounter | f:RiskAssessment/f:encounter | f:ServiceRequest/f:encounter | f:VisionPrescription/f:encounter" />
         <xpathUsage value="normal" />
         <target value="Encounter" />
-        <target value="EpisodeOfCare" />
+        <!-- removed, see: https://github.com/FirelyTeam/fhir-net-api/issues/1298 -->
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Core/Model/Source/search-parameters.xml:13162]: I agree that we should change the source to solve this issue. But should we not also change this file in `Hl7.Fhir.Specification.data`?

Comment (ewoutkramer) [reply src/Hl7.Fhir.Core/Model/Source/search-parameters.xml:13162]: Turns out this specific file only exists in Source (unlike profiles-types and the others).

Comment (mmsmits) [reply src/Hl7.Fhir.Core/Model/Source/search-parameters.xml:13162]: I just removed it

Comment (marcovisserFurore) [reply src/Hl7.Fhir.Core/Model/Source/search-parameters.xml:13162]: Thanks @mmsmits. So `Hl7.Fhir.Specification.data.search-parameter.xml` did exist. But has now been synchronized with `Hl7.Fhir.Core/Model/Source/search-parameters.xml`

**Pull request merged**